python/src/scipp/compat/mantid.py

```diff
@@ -161,9 +161,127 @@ def convert_EventWorkspace_to_dataset(ws, load_pulse_times, EventType):
     return sc.DataArray(**coords_labs_data)
 
 
+def convert_TableWorkspace_to_dataset(ws, error_connection=None):
+    """
+    Converts from a Mantid TableWorkspace to a scipp dataset. A TableWorkspace
+    have optional declaration of connection between data and error using the
+    plottype which can be none, X, Y, Z, Xerr, Yerr. This function collects
+    the data and its error into a single scipp variable if there is no
+    ambiguity in what column contains the error for what data column. This
+    means a maximum of one Xerr and one Yerr column. For more advanced
+    TableWorkspaces, the user can specify the connection between data and error
+    manually using the error_connections keyword argument, which overwrites
+    automatic assignment.
+
+    Parameters
+    ----------
+        ws : Mantid TableWorkspace
+            Mantid TableWorkspace to be converted into scipp dataset
+
+    Keyword arguments
+    -----------------
+        error_connection : Dict
+            Dict with data column names as keys to names of their error column
+    """
+
+    # Extract information from workspace
+    n_columns = ws.columnCount()
+    columnNames = ws.getColumnNames()  # list of names matching each column
+    columnTypes = ws.columnTypes()  # list of types matching each column
```

Comment:
There is `ws.toDict()`, would that be an easier solution for a number of things?

Reply:
I did not find the toDict method easier than the standard access because I wish to do a loop over the columns where some columns should be reassigned to variance. In order to avoid modifying the dictionary being looped, a separate dictionary would have to be allocated regardless.